### PR TITLE
support ipv6 tunnels

### LINKED ISSUE #157: Transport hard-coded for IPv4 (no IPv6 support)
Attempting to connect to an IPv6-only host fails with this error:

```
$ python -c "from sshtunnel import SSHTunnelForwarder; server = SSHTunnelForwarder('myipv6host.example.com', remote_bind_address=('localhost', 8080)); server.start()"                                                     
2019-09-13 14:31:23,039| ERROR   | Could not resolve IP address for myipv6host.example.com, aborting!
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/jaraco/code/public/sshtunnel/sshtunnel.py", line 1294, in start
    reason='Could not establish session to SSH gateway')
  File "/Users/jaraco/code/public/sshtunnel/sshtunnel.py", line 1100, in _raise
    raise exception(reason)
sshtunnel.BaseSSHTunnelForwarderError: Could not establish session to SSH gateway
```

I suspect that's because the [transport is hard-coded for IPv4](https://github.com/pahaz/sshtunnel/blob/397ee78851bdd36d0c11cef319a9f55dd7eb115d/sshtunnel.py#L1114).

It would be nice if the tunnel could connect to an IPv6 host (and prefer that transport based on the results from `getaddrinfo`).

### PR DESCRIPTION
**Pull request by g0djan**
Hi @pahaz , it will fix #157 and #194 issues. We want to use it in our project and I will be glad to fix any troubles with this PR asap.

### CODE REVIEW

File: sshtunnel.py [hunk 1]
```diff
@@ -1101,6 +1101,18 @@ def _raise(self, exception=BaseSSHTunnelForwarderError, reason=None):
...
+        '''
+        This function returns socket according to ip version of ssh_host address
+        '''
+        addrinfo = socket.getaddrinfo(self.ssh_host, self.ssh_port)
+        for info in addrinfo:
+            family = info[0]
+            socktype = info[1]
+            if socktype == socket.SOCK_STREAM:
+                return socket.socket(family, socktype)
+        raise socket.error('Addrinfo for {},{} doesn\'t contain SOCK_STREAM')
```
Comment (pahaz) [inline sshtunnel.py:1114]: It looks like there is a missed `format`. 
I don't think that's it's a good idea to use `socket.error`.

Look at the similar code inside the paramiko: https://github.com/paramiko/paramiko/blob/538caf5/paramiko/client.py#L203-L216

I think it's better to use it.

File: sshtunnel.py [hunk 2]
```diff
@@ -1111,10 +1123,13 @@ def _get_transport(self):
...
-            _socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            _socket = self._get_socket()
         if isinstance(_socket, socket.socket):
             _socket.settimeout(SSH_TIMEOUT)
-            _socket.connect((self.ssh_host, self.ssh_port))
+            address = [self.ssh_host, self.ssh_port]
+            if _socket.family == socket.AF_INET6:
+                address.extend([0, 0])
+            _socket.connect(tuple(address))
         transport = paramiko.Transport(_socket)
```
Comment (pahaz) [inline sshtunnel.py:1133]: At the moment it looks like the paramiko is supporting a tuple (address, port) argument. https://github.com/paramiko/paramiko/blob/dccc563/paramiko/transport.py#L395-L417
Probably it's easy to just pass `(self.ssh_host, self.ssh_port)` as an argument.

**Comment by pahaz** [comment]
Checklist:
 - [ ] write https://github.com/pahaz/sshtunnel/blob/master/changelog.rst#changelog
 - [ ] up package version https://github.com/pahaz/sshtunnel/blob/master/sshtunnel.py#L39
 - [ ] add self to https://github.com/pahaz/sshtunnel/blob/master/changelog.rst#contributors
 - [ ] pull from master

**Comment by g0djan** [comment]
Checked that it works with IPv6 only hosts and still works with IPv4 hosts.

File: changelog.rst [hunk 3]
```diff
@@ -12,10 +12,14 @@ CONTRIBUTORS
 - `Dan Harbin`_
 - `Ignacio Peluffo`_
 - `Niels Zeilemaker`_
+- `Georgy Rylov`_
 
 CHANGELOG
 =========
 
+- v.0.2.0 (`Georgy Rylov`_)
+      + Support IPv6 without proxy command
```
Comment (pahaz) [inline changelog.rst:21]:
```suggestion
      + Support IPv6 without proxy command. Use built-in paramiko create socket logic. The logic tries to use ipv6 socket family first, then ipv4 socket family.
```

**Comment by pahaz** [comment]
Ok. lets ship it

**Review by pahaz — APPROVED** [review]

**Pull request merged**